src/TulsiGenerator/XcodeProjectGenerator.swift

```diff
@@ -414,24 +414,28 @@ final class XcodeProjectGenerator {
                                 intermediateArtifacts: intermediateArtifacts)
   }
 
+  // Resolves the given Bazel path (which is expected to begin with external/) to a filesystem
+  // path. This is intended to be used to resolve "@external_repo" style labels to paths usable by
+  // Xcode. Returns nil if the path could not be resolved for any reason.
+  func resolveExternalReferencePath(_ path: String, xcodeProject: PBXProject) -> String {
+    // Assume that the main group is a relative path to the WORKSPACE directory
+    // When the project is generated at the WORKSPACE root, the external paths should be
+    // `tulsi-workspace/external`. In this case `mainGroup.path` returns nil
+    let relativeWorkspacePath = xcodeProject.mainGroup.path != nil ? "\(xcodeProject.mainGroup.path!)/" : ""
+    return "\(relativeWorkspacePath)\(PBXTargetGenerator.TulsiWorkspacePath)/\(path)"
+  }
+
   // Examines the given xcodeProject, patching any groups that were generated under Bazel's magical
   // "external" container to absolute filesystem references.
   private func patchExternalRepositoryReferences(_ xcodeProject: PBXProject) {
     let mainGroup = xcodeProject.mainGroup
-    guard let externalGroup = mainGroup.childGroupsByName["external"] else { return }
+    guard let externalGroup = mainGroup.childGroupsByName[".."] else { return }
     let externalChildren = externalGroup.children as! [PBXGroup]
```

Comment:
@dmishe ok I've put together a basic example of this, which includes a script to run tulsi and `README` with all of the details. The archive includes a debug build of Tulsi in `tools/Tulsi.app` on this commit ref. Please find the archive here: [tulsi-relative-workspace-example.zip](https://github.com/bazelbuild/tulsi/files/1166305/tulsi-relative-workspace-example.zip)

Just set your `bazel` path in `Scripts/generateXcode.sh` - the script and checked in Xcode project is hardcoded to my `bazel` installation.


<img width="1760" alt="tulsi-relative-source-file-example" src="https://user-images.githubusercontent.com/1245820/28475862-d2a6ece0-6e02-11e7-9428-35332fa628f5.png">


# iOS External Dependency Example

This application is an example of an iOS app with an external dependency.

USAGE:


##  Xcode Project Generation

```
  Scripts/generateXcode.sh
```

The Xcode project, BazelPodPlayground.xcodeproject, has all relative paths for the
source files. The bazel path is hardcoded to whatever "bazel" you are using.
Please set `BAZEL` in  `Scripts/generateXcode.sh`

Note: `external` files will show up red until the symlink `tulsi-workspace` has
been created, which happens during building within Xcode.

If Xcode was opened before the symlink was created, it may need to be restarted.

## Building

```
  bazel build -s //ios-app:ios-app
```

## Info

Built on bazel version:

sha "c31a38761b7b21eadccc757774198bc307353cab07c3c496b386f249de2a33dc"
bazel version url "https://github.com/bazelbuild/bazel/releases/download/0.5.2/bazel-0.5.2-installer-darwin-x86_64.sh"


Tulsi build is based on this pull request:

https://github.com/bazelbuild/tulsi/pull/20/commits/60627f6597f82a952e67bc586aff443f3e3aac2b